testing/trino-testing/src/main/java/io/trino/testing/BaseConnectorTest.java

```diff
@@ -537,6 +537,34 @@ protected void checkInformationSchemaViewsForMaterializedView(String schemaName,
                 .containsAll("VALUES '" + viewName + "'");
     }
 
+    @Test
+    public void testMaterializedAndNormalViewsTogether()
+    {
+        if (!hasBehavior(SUPPORTS_CREATE_MATERIALIZED_VIEW) || !hasBehavior(SUPPORTS_CREATE_VIEW)) {
+            throw new SkipException("Either materialized or normal views not supported");
```

Comment:
fine to `return;` here (see eg `testMaterializedView`)

it's a matter of taste of course. I try to keep amount of skip exceptions low because some reporting tools like to report all the skips (eg my IDE, Jenkins reporting), and then reports become long

Reply:
Fair

Reply:
... bug i agree that "throw skip" is the "more correct", technically speaking